tests/stress.rs

```diff
@@ -1107,23 +1103,58 @@ fn test_dc_get_oauth2_token() {
 }
 
 #[test]
-fn test_stress_tests() {
+fn test_dc_get_abs_path() {
+    let ctx = unsafe { create_test_context() };
+
+    let blobdir_c = unsafe { dc_get_blobdir(&ctx.ctx) };
+    let mut image_path = PathBuf::from(to_string(blobdir_c));
+    image_path.push("image.png");
+
+    let mut image_path_2 = PathBuf::from("$BLOBDIR");
+    image_path_2.push("image.png");
+
+    let image_c = CString::new(image_path_2.to_str().unwrap()).unwrap();
+    let abs_path = unsafe { dc_get_abs_path(&ctx.ctx, image_c.as_ptr()) };
+
+    assert_eq!(to_string(abs_path), image_path.to_str().unwrap());
+}
+
+#[test]
+fn test_selfavatar_config() {
+    let ctx = unsafe { create_test_context() };
+
+    let mut logo_path = current_dir().unwrap();
+    logo_path.push("tests");
+    logo_path.push("fixtures");
+    logo_path.push("delta-logo.png");
+
+    let logo_path_c = CString::new(logo_path.to_str().unwrap()).unwrap();
     unsafe {
-        let context = create_test_context();
-        stress_functions(&context.ctx);
+        dc_set_config(
+            &ctx.ctx,
+            b"selfavatar\x00" as *const u8 as *const libc::c_char,
+            logo_path_c.as_ptr(),
+        );
     }
+
+    let selfavatar = unsafe {
+        dc_get_config(
+            &ctx.ctx,
+            b"selfavatar\x00" as *const u8 as *const libc::c_char,
+        )
+    };
+
+    let blobdir_c = unsafe { dc_get_blobdir(&ctx.ctx) };
+    let mut image_path = PathBuf::from(to_string(blobdir_c));
+    image_path.push("delta-logo.png");
+
+    assert_eq!(to_string(selfavatar), image_path.to_str().unwrap());
```

Comment:
This should fail on windows.

Reply:
no totally sure, but dc_set_config() tries to copy the given file physically to the blob directory. i assume the file described by logo_path_c normally does not exist? maybe the behavior of copying unexistent files is slightly different at some level (dc_set_config() uses the source file when copying failed)

to have a reliable test, maybe ensure that the given file exists before calling dc_get_config()? no need to be a valid image, an empty file should do the job.

Reply:
> to have a reliable test, maybe ensure that the given file exists before calling dc_get_config()? no need to be a valid image, an empty file should do the job.

That's what the test is doing.